docs/csharp/language-reference/operators/conditional-operator.md

```diff
@@ -10,70 +10,76 @@ helpviewer_keywords:
   - "conditional operator (?:) [C#]"
 ms.assetid: e83a17f1-7500-48ba-8bee-2fbc4c847af4
 ---
-
 # ?: Operator (C# Reference)
 
-The conditional operator (`?:`), commonly known as the ternary conditional operator, returns one of two values depending on the value of a Boolean expression. Following is the syntax for the conditional operator.  
+The conditional operator `?:`, commonly known as the ternary conditional operator, evaluates one of two expressions and returns its result depending on the value of a Boolean expression. Beginning with C# 7.2, the [conditional ref expression](#conditional-ref-expression) returns the reference to the result of one of the two expressions.
+
+The syntax for the conditional operator is as follows:
 
 ```csharp
-condition ? first_expression : second_expression;  
+condition ? consequence : alternative
 ```
 
-Beginning with C# 7.2, the `first_expression` and `second_expression` my be [`ref` expressions](https://github.com/dotnet/csharplang/blob/master/proposals/csharp-7.2/conditional-ref.md):
+The `condition` expression must evaluate to `true` or `false`. If `condition` evaluates to `true`, the `consequence` expression is evaluated and its result becomes the result of the operation. If `condition` evaluates to `false`, the `alternative` expression is evaluated and its result becomes the result of the operation. A conditional expression evaluates only one of the two expressions: either `consequence` or `alternative`.
 
-```csharp
-ref condition ? ref first_expression : ref second_expression;  
-```
+The type of `consequence` and `alternative` must be the same, or there must be an implicit conversion from one type to the other.
 
-The result may be assigned to a `ref` or `ref readonly` variable, or to a variable with neither modifier.
+The conditional operator is right-associative, that is, an expression of the form
 
-## Remarks
+```csharp
+a ? b : c ? d : e
+```
 
-The `condition` must evaluate to `true` or `false`. If `condition` is `true`, `first_expression` is evaluated and becomes the result. If `condition` is `false`, `second_expression` is evaluated and becomes the result. Only one of the two expressions is evaluated. This is particularly important for expressions where the result is a `ref`, as the following is valid:
```

Comment:
>This is particularly important for expressions where the result is a `ref`

What makes it so particular important in case of the reference result? If we remove refs from the given example:
```csharp
ref (storage != null) ? ref storage[3] : ref defaultValue;
```

it's also very important that only one expression is evaluated.
So, I've omitted the note about the particular importance; just mention for both expressions that only one expression is evaluated.